src/test/java/com/fundynamic/d2tm/game/event/DrawableViewPortMoverListenerTest.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    @Test
+    public void mouseFromBottomRightToNoBorderStopsMoving() throws SlickException {
+        // this updates viewport coordinates one move up and to the left
+        listener.mouseMoved(ANY_COORDINATE_NOT_NEAR_BORDER, ANY_COORDINATE_NOT_NEAR_BORDER, SCREEN_WIDTH, SCREEN_HEIGHT);
+        updateAndRender(); // and move the viewport
+        Vector2D<Float> tickOneViewportVector = drawableViewPort.getViewingVector();
+
+        // move back to the middle
+        listener.mouseMoved(SCREEN_WIDTH, SCREEN_HEIGHT, ANY_COORDINATE_NOT_NEAR_BORDER, ANY_COORDINATE_NOT_NEAR_BORDER);
+        updateAndRender(); // and stop moving
+
+        // Check that the viewport stopped moving
+        Vector2D<Float> viewportVector = getLastCalledViewport();
+        Assert.assertEquals("Y position moved while expected to have stopped", tickOneViewportVector.getY(), viewportVector.getY(), 0.0001F);
+        Assert.assertEquals("X position moved while expected to have stopped", tickOneViewportVector.getX(), viewportVector.getX(), 0.0001F);
+    }
+
+    @Test
+    public void stopsMovingLeftWhenAtTheLeftEdge() throws SlickException {
+        listener.mouseMoved(0, ANY_COORDINATE_NOT_NEAR_BORDER, 0, ANY_COORDINATE_NOT_NEAR_BORDER); // move left
+        updateAndRender();
+        updateAndRender();
+        updateAndRender();
+
+        Vector2D<Float> viewportVector = getLastCalledViewport();
+        Assert.assertEquals("X position moved over the edge", 0F, viewportVector.getX(), 0.0001F);
+    }
+
+    @Test
+    public void stopsMovingUpWhenAtTheUpperEdge() throws SlickException {
+        listener.mouseMoved(ANY_COORDINATE_NOT_NEAR_BORDER, 0, ANY_COORDINATE_NOT_NEAR_BORDER, 0); // move up
+        updateAndRender();
+        updateAndRender();
+        updateAndRender();
+
+        Vector2D<Float> viewportVector = getLastCalledViewport();
+        Assert.assertEquals("Y position moved over the edge", 0F, viewportVector.getY(), 0.0001F);
+    }
+
+    @Test
+    public void stopsMovingDownWhenAtTheBottomEdge() throws SlickException {
+        int viewportX = 0;
+        int viewportY = 0;
+        float moveSpeed = 16F;
+
+        float maxYViewportPosition = (HEIGHT_OF_MAP * Tile.HEIGHT) - Game.SCREEN_HEIGHT;
+
+        drawableViewPort = new DrawableViewPort(viewport, Vector2D.zero(), new Vector2D<>(viewportX, viewportY), mock(Graphics.class), moveSpeed);
+        listener = new DrawableViewPortMoverListener(drawableViewPort);
+
+        listener.mouseMoved(ANY_COORDINATE_NOT_NEAR_BORDER, SCREEN_HEIGHT, ANY_COORDINATE_NOT_NEAR_BORDER, SCREEN_HEIGHT); // move down
+        updateAndRender();
+        updateAndRender();
+        updateAndRender();
+
+        Vector2D<Float> viewportVector = getLastCalledViewport();
+        Assert.assertEquals("Y position moved over the bottom", maxYViewportPosition, viewportVector.getY(), 0.0001F);
+    }
+
+    @Test
+    public void stopsMovingRightWhenAtTheRightEdge() throws SlickException {
```

Comment:
this test is almost the same as the above one. Refactor? Also, perhaps use `context`s?